Failure to build with Python 3.9.0 on macOS

While rebuilding packages with Python 3.9.0 on macOS Homebrew (https://github.com/Homebrew/homebrew-core/pull/62560), we are seeing a failure to build scipy 1.5.2.

#### Reproducing code example:

```
/usr/local/opt/python@3.9/bin/python3 setup.py build --fcompiler=gnu95
```

#### Error message:

```
scipy/sparse/linalg/dsolve/SuperLU/SRC/clacon2.c:175:5: error: implicit declaration of function 'ccopy_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
scipy/sparse/linalg/dsolve/SuperLU/SRC/dmach.c:62:9: error: implicitly declaring library function 'strncmp' with type 'int (const char *, const char *, unsigned long)' [-Werror,-Wimplicit-function-declaration]
scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_cdrop_row.c:196:3: error: implicit declaration of function 'scopy_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_zdrop_row.c:196:3: error: implicit declaration of function 'dcopy_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
scipy/sparse/linalg/dsolve/SuperLU/SRC/smach.c:62:9: error: implicitly declaring library function 'strncmp' with type 'int (const char *, const char *, unsigned long)' [-Werror,-Wimplicit-function-declaration]
scipy/sparse/linalg/dsolve/SuperLU/SRC/zlacon2.c:175:5: error: implicit declaration of function 'zcopy_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
scipy/sparse/linalg/dsolve/SuperLU/SRC/slacon2.c:160:16: error: implicit declaration of function 'idamax_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
```

--- a/scipy/sparse/linalg/dsolve/SuperLU/SRC/clacon2.c
+++ b/scipy/sparse/linalg/dsolve/SuperLU/SRC/clacon2.c
@@ -106,6 +106,7 @@ clacon2_(int *n, complex *v, complex *x, float *est, int *kase, int isave[3])
     extern float smach(char *);
     extern int icmax1_slu(int *, complex *, int *);
     extern double scsum1_slu(int *, complex *, int *);
+    extern void ccopy_(int *, complex [], int *, complex [], int *);
 
     safmin = smach("Safe minimum");  /* lamch_("Safe minimum"); */
     if ( *kase == 0 ) {

--- a/scipy/sparse/linalg/dsolve/SuperLU/SRC/dmach.c
+++ b/scipy/sparse/linalg/dsolve/SuperLU/SRC/dmach.c
@@ -11,6 +11,7 @@ at the top-level directory.
 #include <float.h>
 #include <math.h>
 #include <stdio.h>
+#include <string.h>
 
 double dmach(char *cmach)
 {

--- a/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_cdrop_row.c
+++ b/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_cdrop_row.c
@@ -23,6 +23,7 @@ at the top-level directory.
 #include <stdlib.h>
 #include "slu_cdefs.h"
 
+extern void scopy_(int *, float [], int *, float [], int *);
 extern void cswap_(int *, complex [], int *, complex [], int *);
 extern void caxpy_(int *, complex *, complex [], int *, complex [], int *);
 extern void ccopy_(int *, complex [], int *, complex [], int *);

--- a/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_zdrop_row.c
+++ b/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_zdrop_row.c
@@ -23,6 +23,7 @@ at the top-level directory.
 #include <stdlib.h>
 #include "slu_zdefs.h"
 
+extern void dcopy_(int *, double [], int *, double [], int *);
 extern void zswap_(int *, doublecomplex [], int *, doublecomplex [], int *);
 extern void zaxpy_(int *, doublecomplex *, doublecomplex [], int *, doublecomplex [], int *);
 extern void zcopy_(int *, doublecomplex [], int *, doublecomplex [], int *);

--- a/scipy/sparse/linalg/dsolve/SuperLU/SRC/slacon2.c
+++ b/scipy/sparse/linalg/dsolve/SuperLU/SRC/slacon2.c
@@ -100,10 +100,12 @@ slacon2_(int *n, float *v, float *x, int *isgn, float *est, int *kase, int isave
     float temp;
 #ifdef _CRAY
     extern int ISAMAX(int *, float *, int *);
+    extern int IDAMAX(int *, float *, int *);
     extern float SASUM(int *, float *, int *);
     extern int SCOPY(int *, float *, int *, float *, int *);
 #else
     extern int isamax_(int *, float *, int *);
+    extern int idamax_(int *, float *, int *);
     extern float sasum_(int *, float *, int *);
     extern int scopy_(int *, float *, int *, float *, int *);
 #endif

--- a/scipy/sparse/linalg/dsolve/SuperLU/SRC/smach.c
+++ b/scipy/sparse/linalg/dsolve/SuperLU/SRC/smach.c
@@ -11,6 +11,7 @@ at the top-level directory.
 #include <float.h>
 #include <math.h>
 #include <stdio.h>
+#include <string.h>
 
 float smach(char *cmach)
 {

--- a/scipy/sparse/linalg/dsolve/SuperLU/SRC/sp_ienv.c
+++ b/scipy/sparse/linalg/dsolve/SuperLU/SRC/sp_ienv.c
@@ -68,6 +68,7 @@ at the top-level directory.
 int
 sp_ienv(int ispec)
 {
+    extern int input_error(char *, int *);
     int i;
 
     switch (ispec) {

--- a/scipy/sparse/linalg/dsolve/SuperLU/SRC/zlacon2.c
+++ b/scipy/sparse/linalg/dsolve/SuperLU/SRC/zlacon2.c
@@ -106,6 +106,7 @@ zlacon2_(int *n, doublecomplex *v, doublecomplex *x, double *est, int *kase, int
     extern double dmach(char *);
     extern int izmax1_slu(int *, doublecomplex *, int *);
     extern double dzsum1_slu(int *, doublecomplex *, int *);
+    extern void zcopy_(int *, doublecomplex [], int *, doublecomplex [], int *);
 
     safmin = dmach("Safe minimum");  /* lamch_("Safe minimum"); */
     if ( *kase == 0 ) {

